tests/linear_model/test_glm.py

```diff
@@ -203,3 +219,121 @@ def test_logistic_predict_proba_shape():
+    np_coef = np.hstack((np_mod.coef_, np_mod.intercept_))
+
+    rel_error = LA.norm(da_coef - np_coef) / LA.norm(np_coef)
+    assert rel_error < 1e-8
+
+
+@pytest.mark.parametrize("solver", ["newton", "gradient_descent", "lbfgs"])
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize(
+    "reg_type, skl_reg_type, skl_params, data_generator",
+    [
+        (LinearRegression, slm.LinearRegression, {}, "medium_size_regression"),
+        (
+            LogisticRegression,
+            slm.LogisticRegression,
+            {"penalty": "none"},
+            "single_chunk_classification",
+        ),
+        (
+            PoissonRegression,
+            slm.PoissonRegressor,
+            {"alpha": 0},
+            "medium_size_counts",
+        ),
+    ],
+)
+def test_model_against_sklearn(
+    reg_type, skl_reg_type, skl_params, data_generator, fit_intercept, solver, request
+):
+    """
+    Tests accuracy of model predictions
+    """
+    X, y = request.getfixturevalue(data_generator)
+    dask_ml_model = reg_type(
+        fit_intercept=fit_intercept, solver=solver, penalty="l2", C=1e8, max_iter=500
+    )
+    skl_model = skl_reg_type(fit_intercept=fit_intercept, **skl_params)
+    # skl_model has to be fit with numpy data
+    dask_ml_model.fit(X, y)
+    skl_model.fit(X.compute(), y.compute())
+
+    est, truth = dask_ml_model.predict(X), skl_model.predict(X)
+    rel_error = LA.norm(est - truth) / LA.norm(truth)
+    assert rel_error < 1e-3
+
+
+@pytest.mark.parametrize("solver", ["newton", "gradient_descent", "lbfgs"])
+@pytest.mark.parametrize(
+    "reg_type, skl_reg_type, skl_params, dataset_function",
+    [
+        (LinearRegression, slm.LinearRegression, {}, make_regression),
+        (
+            LogisticRegression,
+            slm.LogisticRegression,
+            {"penalty": "none"},
+            make_classification,
+        ),
+    ],
+)
+def test_model_coef_against_sklearn(
+    reg_type, skl_reg_type, skl_params, dataset_function, solver
+):
+    """
+    Tests accuracy of coefficient calculation.
+
+    sk-learn least squares method does not tie out to iterative solvers with
+    e.g. Rdi
+    """
+    X, y = dataset_function(
+        n_samples=100, n_features=20, n_informative=10, chunks=100, random_state=0
+    )
+    X_np, y_np = X.compute(), y.compute()
+
+    skl_model = skl_reg_type(fit_intercept=True, **skl_params)
+    skl_model.fit(X_np, y_np)
+
+    dask_ml_model = reg_type(fit_intercept=True, solver=solver, penalty="l2", C=1e8)
+    dask_ml_model.fit(X, y)
+
+    est, truth = np.hstack((dask_ml_model.intercept_, dask_ml_model.coef_)), np.hstack(
+        (skl_model.intercept_, skl_model.coef_.flatten())
+    )
+    rel_error = LA.norm(est - truth) / LA.norm(truth)
+    assert rel_error < 1e-3
+
+
+@pytest.mark.parametrize("solver", ["newton", "gradient_descent", "lbfgs"])
+def test_poisson_regressor_against_sklearn(single_chunk_count_classification, solver):
```

Comment:
Why isn't this test part of `test_model_coef_against_sklearn`?

Reply:
Combined into a single test.